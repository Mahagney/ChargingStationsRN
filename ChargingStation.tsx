import { StyleSheet, Text, View } from "react-native";
import Logo from "./assets/RightArrow.svg"
import Star from "./assets/star.svg";

const calculateDistance = (distance: number) => {
    return distance>1000 ? `${distance/1000}km` : `${distance}m`;
}

export type ChargingPointType = {
    "availableChargePoints": string, 
    "distance": string, 
    "isFavourite": boolean, 
    "location": string, 
    "stationName": string, 
    "totalChargePoints": string
}

export const ChargingStation = ({availableChargePoints, totalChargePoints, stationName, location, isFavourite, distance, isLast, isFirst}:ChargingPointType & {isLast:boolean, isFirst: boolean}) => {
    
    const style = StyleSheet.create({
        container: {
            display: 'flex',
            flexDirection: 'row',
            width: '100%',
            height: 70,
            borderBottomWidth: isLast ? 0 : 1,
            borderBottomColor: '#b8bfc4',
        },
        distanceContainer: {
            display: 'flex',
            flexDirection: 'column',
            alignItems:'center',
            backgroundColor: '#e6ebee',
            padding: 16,
            overflow: 'hidden',
            width: 80,
            zIndex: 2,
            borderTopRightRadius: isFirst ? 10 : 0,
            borderBottomRightRadius: isLast ? 10 : 0
        },
    });
    return (
        <View style={style.container}>
            <View style={styles.availableContainer}>
                <Text style={styles.totalText}> {availableChargePoints}/{totalChargePoints} </Text>
                <Text> Available </Text>
            </View>
            <View style={styles.addressContainer}>
                <View style={styles.starContainer}>
                    <Text style={ styles.addressText}>{ stationName.length >15 ? `${stationName.slice(0,15)}...` : stationName } </Text>
                    {isFavourite && <Star style={styles.star}/>}
                </View>
                <Text style={ styles.addressText}>{ location.length >18 ? location.slice(0,18) : location }</Text>
            </View>
            <View style={style.distanceContainer}>
                <Logo/>
                <Text>{calculateDistance(parseInt(distance))}</Text>    
            </View>            
        </View>
    );
}

const styles = StyleSheet.create({
    availableContainer: {
        display: 'flex',
        flexDirection: 'column',
        padding: 16,
    },
    starContainer: {
        display: 'flex',
        flexDirection:'row'
    },
    star: {
        marginTop: 4
    },
    addressContainer: {
        display: 'flex',
        flexDirection: 'column',
        padding: 16,
        flexGrow: 1
    },
    totalText: {
        fontWeight: '700',
        fontSize: 18,
    },
    addressText: {
        fontSize: 12,
        color: '#383d3f',
    }
});