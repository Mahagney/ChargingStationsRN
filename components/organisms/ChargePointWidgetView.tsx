/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */
import React, {useEffect, useState} from 'react';
import {
  ScrollView,
  StyleSheet,
  Text,
  View,
} from 'react-native';

import { ChargingStation, ChargingPointType } from '../molecules/ChargingStation.tsx';
import { SHOW_ALL, SHOW_LESS } from '../../constants.ts';
import { Button } from '../atoms/button.tsx';


export const ChargePointWidgetView = ({chargePoints}: {chargePoints:ChargingPointType[]}): React.JSX.Element  => {
  const [showAll, setShowAll] = useState<boolean>(false);
  const [displayedChargePoints, setDisplayedChargePoints] = useState<ChargingPointType[]>([]);
  
  useEffect(() => {
    setDisplayedChargePoints(chargePoints);
}, [chargePoints])

  const toggleShowAll = () => {  
    if(showAll) {
      setDisplayedChargePoints(chargePoints);
    } else {
      setDisplayedChargePoints(chargePoints.slice(0,3));
    }
    setShowAll(!showAll);
  }

  const generateChargingStations = () => {
    return displayedChargePoints.map( (cp: ChargingPointType,index: number) =>             
    <ChargingStation key={index} isLast={index===displayedChargePoints.length-1}
      {...cp} isFirst={index===0}/>
    )
  };

  return (
      <ScrollView   style={styles.container} contentContainerStyle={{paddingBottom: 100, backgroundColor: '#ecf0f3'}}> 
            <Text style={styles.hello}>Hello</Text>
            <Text style={styles.userName}>Saleh</Text>
            <Text style={styles.title}>Nearby Chargers</Text>
            <Text style={styles.text}>Nearest charging stations within a radius of 50km </Text>
            <Text style={styles.text}>Your Preference: CCS Type 2</Text>
            <View style={styles.view}>
              {generateChargingStations()}
            </View>
            <Button text={showAll ? SHOW_ALL : SHOW_LESS} onClick={toggleShowAll}/>
      </ScrollView>
  );
}

const styles = StyleSheet.create({
  hello:{
    fontSize: 18,
    fontWeight: '500'
  },
  userName:{
    marginTop: 8,
    fontSize: 36
  },
  title: {
    marginTop: 40,
    fontWeight: '700'
  },
  subtitle: {
    color: '#ecf0f3'
  },
  text: {
    color: '#616e76',
    fontSize: 12  
  },
  container: {
    display: 'flex',
    flexGrow: 1,
    paddingVertical: 64,
    paddingHorizontal: 16,
    backgroundColor: '#ecf0f3',
    minHeight: '100%'
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
  view: {
    zIndex: 5,
    borderRadius: 10,
    borderColor: "#a3b8c5",
    borderWidth: 1,
    marginVertical: 16,
    backgroundColor: 'white',
    flexDirection: "column",
    display: 'flex',
  }
});

export default ChargePointWidgetView;
