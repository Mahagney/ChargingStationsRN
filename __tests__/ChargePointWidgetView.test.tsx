import {ReactTestRenderer, act, create} from 'react-test-renderer';
import { ChargingPointType } from '../components/molecules/ChargingStation.tsx';
import {expect, it} from '@jest/globals';
import ChargePointWidgetView from '../components/organisms/ChargePointWidgetView.tsx';

jest.mock('../assets/star.svg', () => () => null) 
jest.mock('../assets/RightArrow.svg', () => () => null) 

const data: ChargingPointType[] = [
{"availableChargePoints": "5", "distance": "780", "isFavourite": true, "location": "LE11 EC3, UK", "stationName": "Loughborough towncentre hall2", "totalChargePoints": "5"}, 
{ "availableChargePoints": "9", "distance": "140", "isFavourite": true, "location": "W2 5TY, UK", "stationName": "Paddington 36", "totalChargePoints": "18"}, 
{ "availableChargePoints": "3", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 2", "totalChargePoints": "10"}, 
{"availableChargePoints": "4", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 4", "totalChargePoints": "10"}, 
{"availableChargePoints": "5", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 5", "totalChargePoints": "10"}, 
{"availableChargePoints": "8", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 6", "totalChargePoints": "10"}, 
{ "availableChargePoints": "1", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 8", "totalChargePoints": "10"}];
// render the component

it('renders correctly', async () => {
  let renderer: ReactTestRenderer | undefined = undefined;
  act(() => {
    renderer = create(<ChargePointWidgetView chargePoints={data}/>);//.toJSON();
});
  expect(renderer).not.toBeNull;

  expect(renderer).toMatchSnapshot();

  const button = renderer!.root!.findByProps({testID: "show"});
  expect(button).not.toBeFalsy;
  act(() => {button.props.onPress();});

  expect(renderer).toMatchSnapshot();

});