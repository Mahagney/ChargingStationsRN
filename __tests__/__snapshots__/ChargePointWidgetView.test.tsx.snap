// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "backgroundColor": "#ecf0f3",
      "paddingBottom": 100,
    }
  }
  style={
    {
      "backgroundColor": "#ecf0f3",
      "display": "flex",
      "flexGrow": 1,
      "minHeight": "100%",
      "paddingHorizontal": 16,
      "paddingVertical": 64,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
    >
      Hello
    </Text>
    <Text
      style={
        {
          "fontSize": 36,
          "marginTop": 8,
        }
      }
    >
      Saleh
    </Text>
    <Text
      style={
        {
          "fontWeight": "700",
          "marginTop": 40,
        }
      }
    >
      Nearby Chargers
    </Text>
    <Text
      style={
        {
          "color": "#616e76",
          "fontSize": 12,
        }
      }
    >
      Nearest charging stations within a radius of 50km 
    </Text>
    <Text
      style={
        {
          "color": "#616e76",
          "fontSize": 12,
        }
      }
    >
      Your Preference: CCS Type 2
    </Text>
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#a3b8c5",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "column",
          "marginVertical": 16,
          "zIndex": 5,
        }
      }
    >
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            5
            /
            5
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              Loughborough to...
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            LE11 EC3, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 10,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            780m
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            9
            /
            18
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              Paddington 36
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            W2 5TY, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            140m
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            3
            /
            10
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              London 2
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            W2 5TY, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            12km
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            4
            /
            10
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              London 4
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            W2 5TY, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            12km
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            5
            /
            10
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              London 5
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            W2 5TY, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            12km
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            8
            /
            10
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              London 6
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            W2 5TY, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            12km
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 0,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            1
            /
            10
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              London 8
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            W2 5TY, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 10,
              "borderTopRightRadius": 0,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            12km
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#002899",
          "borderRadius": 20,
          "borderWidth": 1,
          "paddingVertical": 8,
        }
      }
      testID="show"
    >
      <Text
        style={
          {
            "textAlign": "center",
          }
        }
      >
        Show less
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders correctly 2`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "backgroundColor": "#ecf0f3",
      "paddingBottom": 100,
    }
  }
  style={
    {
      "backgroundColor": "#ecf0f3",
      "display": "flex",
      "flexGrow": 1,
      "minHeight": "100%",
      "paddingHorizontal": 16,
      "paddingVertical": 64,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
    >
      Hello
    </Text>
    <Text
      style={
        {
          "fontSize": 36,
          "marginTop": 8,
        }
      }
    >
      Saleh
    </Text>
    <Text
      style={
        {
          "fontWeight": "700",
          "marginTop": 40,
        }
      }
    >
      Nearby Chargers
    </Text>
    <Text
      style={
        {
          "color": "#616e76",
          "fontSize": 12,
        }
      }
    >
      Nearest charging stations within a radius of 50km 
    </Text>
    <Text
      style={
        {
          "color": "#616e76",
          "fontSize": 12,
        }
      }
    >
      Your Preference: CCS Type 2
    </Text>
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#a3b8c5",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "column",
          "marginVertical": 16,
          "zIndex": 5,
        }
      }
    >
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            5
            /
            5
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              Loughborough to...
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            LE11 EC3, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 10,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            780m
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            9
            /
            18
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              Paddington 36
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            W2 5TY, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 0,
              "borderTopRightRadius": 0,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            140m
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#b8bfc4",
            "borderBottomWidth": 0,
            "display": "flex",
            "flexDirection": "row",
            "height": 70,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "padding": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "700",
              }
            }
          >
             
            3
            /
            10
             
          </Text>
          <Text>
             Available 
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 1,
              "padding": 16,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#383d3f",
                  "fontSize": 12,
                }
              }
            >
              London 2
               
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#383d3f",
                "fontSize": 12,
              }
            }
          >
            W2 5TY, UK
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e6ebee",
              "borderBottomRightRadius": 10,
              "borderTopRightRadius": 0,
              "display": "flex",
              "flexDirection": "column",
              "overflow": "hidden",
              "padding": 16,
              "width": 80,
              "zIndex": 2,
            }
          }
        >
          <Text>
            12km
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#002899",
          "borderRadius": 20,
          "borderWidth": 1,
          "paddingVertical": 8,
        }
      }
      testID="show"
    >
      <Text
        style={
          {
            "textAlign": "center",
          }
        }
      >
        Show all
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
